CTFd/models.py

```diff
@@ -159,20 +159,34 @@ def __init__(self, name, email, password):
     def __repr__(self):
         return '<team %r>' % self.name
 
-    def score(self):
+    def score(self, date_filter=None):
         score = db.func.sum(Challenges.value).label('score')
-        team = db.session.query(Solves.teamid, score).join(Teams).join(Challenges).filter(Teams.banned == False, Teams.id == self.id).group_by(Solves.teamid).first()
+        team = db.session.query(Solves.teamid, score).join(Teams).join(Challenges).filter(Teams.banned == False, Teams.id == self.id)
         award_score = db.func.sum(Awards.value).label('award_score')
-        award = db.session.query(award_score).filter_by(teamid=self.id).first()
+        award = db.session.query(award_score).filter_by(teamid=self.id)
+
+        if date_filter:
```

Comment:
Perhaps it's better to automatically grab the freeze time from the db instead of having it passed in?